### PR TITLE
fix(FormatURLApiModule): decode percent-encoding exactly once

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/XSUP-76731

## Description
URLCheck looped unquote until the value stopped changing. That's lossy: %0D%0A became a literal CRLF (splitting one URL indicator across three lines). which leads to such an url indicator
```
example.com/?unsubscribe.
info@example.com
```

## Screenshot
Before
Screenshot 2026-09-15 at 22 07 54
After
Screenshot 2026-09-15 at 22 20 32

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
🚨 **Security Review Required**

This PR triggered security alerts that need reviewer attention.

@DeanArbel, @github-code-quality[bot], @kamalq97, @MosheEichler — this review **cannot be approved** until the security findings are resolved.

Please contact infra team for more details.

**Review by kamalq97 — APPROVED** [review]

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/ApiModules/Scripts/FormatURLApiModule/FormatURLApiModule.py [hunk 1]
```diff
@@ -52,6 +53,24 @@ class URLCheck:
...
+        Control characters are never legal raw in a URL, so re-encoding them is always safe.
+
+        Args:
+            url: The URL after percent-decoding
+
+        Returns:
+            The URL with every control character percent-encoded again
+        """
+
+        return "".join(f"%{ord(char):02X}" if unicodedata.category(char) == "Cc" else char for char in url)
```
Comment (marketplace-ai-reviewer) [inline Packs/ApiModules/Scripts/FormatURLApiModule/FormatURLApiModule.py]: Consider using `urllib.parse.quote(char)` instead of `f"%{ord(char):02X}"` to handle non-ASCII control characters correctly.

```python
        return "".join(urllib.parse.quote(char) if unicodedata.category(char) == "Cc" else char for char in url)
```

File: Packs/ApiModules/Scripts/FormatURLApiModule/FormatURLApiModule_test.py [hunk 2]
```diff
@@ -607,6 +607,93 @@ def test_remove_trailing_bracket_and_redundant_characters_from_part(
...
+        assert URLFormatter(url_).__str__() == expected
+
+    @pytest.mark.parametrize(
+        "url_",
+        [
+            CRLF_INJECTION[0],
+            "https://test.com/path?a=1%0D%0Ahttps://evil.com",
+            "https://test.com/path?a=1%0d%0avictim@example.com",
+            "https://test.com/path?a=1%0Abreak",
+            "https://test.com/path?a=1%09tab",
```
Comment (marketplace-ai-reviewer) [inline Packs/ApiModules/Scripts/FormatURLApiModule/FormatURLApiModule_test.py:660]: Consider adding a test case with a null byte (`%00`) to ensure all types of control characters are verified.

```python
            "https://test.com/path?a=1%09tab",
            "https://test.com/path?a=1%00null",
```

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for this contribution! I've reviewed your PR and have a few quick suggestions to improve the metadata and URL handling. Please ensure the pack metadata includes at least one category and the vendor name in the keywords. For the URL formatting module, consider using `urllib.parse.quote` for safer control character handling and expanding your tests to cover null bytes. Great work overall!

Additionally, please address the following file-level notes:
- `Packs/ApiModules/pack_metadata.json`: - The mandatory key `categories` is present but empty. Please ensure it is populated with at least one valid category.
- The vendor name is missing from the `keywords` list. Please add 'Cortex' or 'XSOAR' to the `keywords`.

 @kamalq97, @marketplace-ai-reviewer, @github-code-quality[bot], @DeanArbel, @MosheEichler please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by MosheEichler** [comment]
The PR needs a force merge as if [fails](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/jobs/78722951#L1636) on a [known](https://jira-dc.paloaltonetworks.com/browse/CIAC-11720) test failure
